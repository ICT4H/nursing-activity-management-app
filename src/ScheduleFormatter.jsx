import React, {Component} from 'react';
import {ADMINISTRATED, NOTADMINISTRATED, TOBEADMINISTRATED} from "./constants";

function FormatTime(time) {
  //AM and PM not showing currently (24 hours clock);
  return <p>{time.getHours() + ":" + time.getMinutes()}</p>;
}

class ScheduleFormatter extends Component {
  render() {
    let schedule=this.props.schedule;
    return (<div className="schedule">
      {FormatTime(schedule.scheduledTime)} {this.getScheduleStatus()}
    </div>)
  }

  //3 ifs looks very bad
  getScheduleStatus() {
    if(this.props.schedule.status===ADMINISTRATED){
      return <span>&#x2705;</span>;
    }
    if(this.props.schedule.status===TOBEADMINISTRATED){
      return <span>&#x23F0;</span>
      // return <p>&#x23F1;</p>;
    }
    if(this.props.schedule.status===NOTADMINISTRATED){
            return <span>&#x26A0;</span>
    }
  }
}

export default ScheduleFormatter
