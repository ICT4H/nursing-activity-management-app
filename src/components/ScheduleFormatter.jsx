import React, {Component} from 'react';
import {SYMBOLSHEXCODE} from "../constants";
import DateUtils from "../utils/DateUtils";

class ScheduleFormatter extends Component {
  render() {
    let schedule = this.props.schedule;
    return (<div className="schedule">
      <span>
        <span className="scheduleTime">{DateUtils.formatTime(schedule.scheduledTime)}</span>
        {this.getScheduleStatus()}</span>
    </div>)
  }

  getScheduleStatus() {
    let statusHexCode = SYMBOLSHEXCODE[this.props.schedule.status];
    return <span className="scheduleStatus">{String.fromCharCode(statusHexCode)}</span>;
  }
}

export default ScheduleFormatter
