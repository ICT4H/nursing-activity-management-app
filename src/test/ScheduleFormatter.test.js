import React from 'react';
import ScheduleFormatter from '../ScheduleFormatter';
import renderer from 'react-test-renderer';

test('Should have time of given schedule', () => {
  let schedule = {scheduledTime:new Date("June 8, 2018 2:30:00")};
  const component = renderer.create(
      <ScheduleFormatter schedule={schedule}/>,
  );
  expect(component.root.findByType("p").props.children).toBe("2:30");
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


//Need to write test for getting emoji based on schedule status