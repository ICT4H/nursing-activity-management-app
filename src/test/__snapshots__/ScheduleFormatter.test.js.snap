// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should have time of given schedule 1`] = `
<div
  className="schedule"
>
  <p>
    2:30
  </p>
   
</div>
`;
